jetty-server/src/main/java/org/eclipse/jetty/server/session/SessionHandler.java

```diff
@@ -328,10 +331,7 @@ public void addEventListener(EventListener listener)
      */
     public void clearEventListeners()
     {
-        for (EventListener e : getBeans(EventListener.class))
-        {
-            removeBean(e);
-        }
+        // TODO should we remove from super?
```

Comment:
Yes, any beans that were added via addEventListener must be removed from super as well. When we parse web.xml we gather all the listeners, call SessionHandler.clearEventListeners() then call SessionHandler.setEventListeners().